service/history/historyEngine.go

```diff
@@ -2584,6 +2584,10 @@ func (e *historyEngineImpl) SignalWithStartWorkflowExecution(
 				return nil, workflow.ErrSignalsLimitExceeded
 			}
 
+			if requestID := sRequest.GetRequestID(); requestID != "" {
```

Comment:
How did you notice this was missing? It's a little bit unclear to me what are the side effects of doing this here. There's already `AddWorkflowExecutionSignaled` below, do these get deduped somehow? 
It would be great if we take this opportunity to add some extra documentation/diagram for SignalWithStart flow.

Reply:
I tested it locally and found it's not idempotent in this case.